Lib/_pyio.py

```diff
@@ -504,8 +504,11 @@ def nreadahead():
                 return 1
         if size is None:
             size = -1
-        elif not isinstance(size, int):
-            raise TypeError("size must be an integer")
+        else:
+            try:
+                size = size.__index__()
+            except AttributeError as err:
+                raise TypeError("an integer is required") from err
```

Comment:
when I remove the reassignment, I get `TypeError: '<' not supported between instances of 'int' and 'IntLike'` somewhere later, when the size var is compared to an int.
Did I misunderstand your comment?